freedomofpress/securedrop-client
# data migration for journalist association fix in draftreplies table

Fixes #1185

# Description

Just like we did for the `replies` table, apply a data migration on the `draftreplies` table, see https://github.com/freedomofpress/securedrop-client/pull/1162 for context.

# Test Plan

1. Checkout the 0.2.1 client (https://github.com/freedomofpress/securedrop-client/releases/tag/0.2.1) 
2. `pip install --require-hashes -r dev-requirements.txt`
3. Send some failed replies
   - `wget https://gist.githubusercontent.com/creviera/5ba70d50c12b6a6df6f98ed40ad09645/raw/5caef3339ceab1fc997ccb6b9e337bc8828ef12f/failed-reply`
   - `git apply failed-reply`
   - Run the client and send some replies
     - [ ] Confirm that the `journalist_id` column of `draftreplies` table contains uuids for the failed draftreplies you just tried to send
4. Now checkout this PR branch
5. `pip install --require-hashes -r requirements/dev-requirements.txt`
6. At this point, if you run the client, and view the failed replies, you will notice two things:
   - [ ] failed replies are missing the initials of the journalist in the badges
   - [ ] the client will crash after sync
7. Now, apply the migration (this is done in the postinst which we will test during QA, but for now, follow the steps below)
   - first make sure `alembic.ini` points to the `svs.sqlite` db with your failed replies (edit this manually: I had to update the sqlalchemy.url  to `sqlalchemy.url = sqlite:////home/creviera/.securedrop_client/svs.sqlite`)
   - now, stamp the db to the previous revision: `alembic stamp 7f682532afa2`
   - now, apply the migration: `alembic upgrade head`
8. Run the client
   - [ ] Verify failed replies now have the initials of the journalist in the badges
   - [ ] Verify the client no longer crashes after sync
   - [ ] Confirm that the `journalist_id` column of `draftreplies` table no longer contains uuids

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:


If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:


If these changes modify the database schema, you should include a database migration. Please check as applicable:
